x/concentrated-liquidity/swaps.go

```diff
@@ -478,17 +475,22 @@ func (k Keeper) calcInAmtGivenOut(
 
 		// utilizing the bucket's liquidity and knowing the price target, we calculate the how much tokenOut we get from the tokenIn
 		// we also calculate the swap state's new sqrtPrice after this swap
-		sqrtPrice, amountIn, amountOut := swapStrategy.ComputeSwapStep(
+		sqrtPrice, amountOut, amountIn := swapStrategy.ComputeSwapStep(
```

Comment:
AFAIR, doesnt this function return amountIn and out, respectively?(implying that the order we used before was correct?)

Reply:
In the context of swap out given in you would be right.

However, here everything is reversed since now we are giving amount out as an input to this function.

We need to refactor `ComputeSwapStep` to be more generalized with the names of the internal variables and its returns.

I left it out of this PR as it is out of scope. I will add this as TODO for https://github.com/osmosis-labs/osmosis/issues/3218